python/modules/connection.py

```diff
@@ -16,9 +16,9 @@
 async def send_offer(msg, agent):
     endpoint = msg.message['endpoint']
     name = msg.message['name']
+    nonce = agent.ui_token
```

Comment:
A unique `nonce` *MUST* be generated for every connection. The `name` isn't a secure enough identifier to use in place of a `nonce`.

This `nonce` is used in multiple contexts. At the most basic level that we're working on right now with agents, the `nonce` only proves that the request identified by that `nonce` is actually one that we solicited. When you introduce Agencies and Cloud Agents, however, the need for the `nonce` increases dramatically. The `nonce` is used in this context to route messages from the agency to the correct cloud agent and then on to the correct edge agent.

Hopefully we can make this process clearer as we produce the HIPE I mentioned in a previous comment detailing how connections are established between agents.